inst/training_202307/202307.qmd

```diff
@@ -0,0 +1,448 @@
+---
+title: "`verdepcheck`"
+author: "Pawel Rucki"
+date: 2023/08/07
+format: revealjs
+---
+
+## Agenda
+
+1. Motivation
+1. Solution
+1. How to use?
+1. How it works?
+1. Discussion
+1. Future work
+
+# Motivation
+
+```{.yaml code-line-numbers="7"}
+Type: Package
+Package: foo.package
+Version: 1.2.3.9000
+Depends:
+    R (>= 3.6)
+Imports:
+    dplyr,
+    (...)
+```
+
+The above specification indicates compatibility with **all** versions of `dplyr` which is very unlikely to be true.
+
+# Motivation
+
+```.default
+Error: object ‘foo’ is not exported by 'namespace:bar'
+```
+
+<br>
+
+```.default
+`foo()` was deprecated in bar 1.0.0.
+i Please use `baz()` instead.
+```
+
+<br>
+
+The new versions of dependencies might introduce a breaking changes.
+
+# Motivation
+
+R environments for clinical trials are usually not updated very frequently.
+
+Users might use `renv` for environment management and packages used might not get updates for some time.
+
+As a result, your package might be used with _older_ version of dependent packages.
+
+# Solution
+
+`verdepcheck` (**ver**sion of **dep**endencies **check**) - a tool for package developers to check your package against different set of dependencies. It is meant to be used primarily in CI.
+
+Available at:
+
+- [`insightsengineering/verdepcheck`](https://github.com/insightsengineering/verdepcheck)
+- [`insightsengineering/r-verdepcheck-action`](https://github.com/insightsengineering/r-verdepcheck-action)
+
+:::{.callout-important}
+Do not confuse `verdepcheck` with `revdepcheck` for reverse dependency checks. Those are totally different tools!
+:::
+
+#
+
+Let's call a "different set of dependencies" a _strategy_. Currently there are three _strategies_ implemented:
+
+- **max** - use _development_ version of dependencies
```

Comment:
are min and release really a searches like min is, or are these simply  development and prod cohort check? if so, it doesn't doesn't seem like they belong in the same discussion as min.

Reply:
Good point. Both release and max don't include version search similar to the one that exists in the min strategy. If it makes it easier to understand the scope I can try do differentiate it more.
I have to admit that at the very beginning I was having a little difficulties with how to conceptually combine those three functionalities under a single product. Both release and max (as we name it right now), even if they are relatively simpler, still give some valuable insights so wanted to keep them here. The common ground here is a "set of dependencies" and my idea was then to call it a _strategy_. Then IMHO those three algorithms fits nicely into that definition.